### ISSUE TITLE
problem with rotate and perspective module and ctl-z

### ISSUE THREAD

**Issue by paolobenve** [darktable-org/darktable#10904]
3.8

In darkroom mode:
- show **rotate and perspective** module
- click on automatically correct for vertical perspective distorsion or on analyze line structure in image
- ctl-z
- click again on automatically correct for vertical perspective distorsion or on analyze line structure in image

-> you get "data pending - please repeat"

moving back to previous image which has the perspective correction and the crop and rotate modules applied, you get it without cropping

Heritage of perspective module, see #10294

**Comment by Nilvus** [comment]
@AlicVB: I know you like when we give you more work ;-)

**Comment by AlicVB** [comment]
yes, I'm aware of this, but I've failed to solve it until now... Note that you "just" have to focus another module and return to r&p to "solve" it, no need to go to another image (that doesn't mean it's not a bug...)

**Comment by paolobenve** [comment]
> you "just" have to focus another module and return to r&p to "solve" it, no need to go to another image 

no, it doesn't work for me

**Comment by AlicVB** [comment]
strange that it doesn't work for you... another one ;)

to be sure here is the steps I've done : 
1. open fresh image (no xmp...)
2. go to r&p module (by clicking on the header to expand it)
3. click on the "analyse structure" button
4. ctrl-z
5. click another time on the "analyze structures" btn => get the "data pending" message
6. go to "orientation" module (by clicking on the header to expand it) This collapse the r&p module
7. go back to r&p module (by clicking on the header to expand it)
8. click on "analyse structures" => works as intended

note that on step 8, you have in fact to click 2 times, as the first one just "untoggle" the button

**Comment by paolobenve** [comment]
@AlicVB you are right
